http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2PingFrameHandler.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal.http2;
+
+import static software.amazon.awssdk.http.nio.netty.internal.ChannelAttributeKey.PING_STATUS;
+
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.SimpleChannelInboundHandler;
+import io.netty.handler.codec.http2.Http2PingFrame;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.utils.Logger;
+
+@SdkInternalApi
+@ChannelHandler.Sharable
+public final class Http2PingFrameHandler extends SimpleChannelInboundHandler<Http2PingFrame> {
+    private static final Logger log = Logger.loggerFor(Http2PingFrameHandler.class);
+    private static final Http2PingFrameHandler INSTANCE = new Http2PingFrameHandler();
+
+    private Http2PingFrameHandler() {
+    }
+
+    public static Http2PingFrameHandler create() {
+        return INSTANCE;
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, Http2PingFrame frame) {
+        if (frame.ack()) {
+            log.debug(() -> "Received PING ACK from channel " + ctx.channel());
+            ctx.channel().attr(PING_STATUS).set(PingStatus.ACK_RECEIVED);
```

Comment:
What happens if:
[0:00] We get a ReadTimeout
[0:00] We send a PING request (state = IN_FLIGHT)
[0:00] We set a timer to check for the PING response after 5 seconds
[0:01] We get the PING response (state = ACK_RECEIVED)
[0:04] We get a ReadTimeout
[0:04] We send a PING request (state = IN_FLIGHT)
[0:05] The timer triggers and sees that the PING is IN_FLIGHT

Reply:
Good call.. we need to cancel the timer once it's received